benchmarks/htap/lib/transactions.py

```diff
@@ -0,0 +1,163 @@
+        rbk = self.random.randint_inclusive(1, 100)
+        itemid = []
+        supware = []
+        qty = []
+        all_local = 1
+
+        for order_line in range(1, order_line_count + 1):
+            itemid.append(self.random.nurand(8191, 1, MAXITEMS))
+            if (order_line == order_line_count - 1) and (rbk == 1):
+                itemid[-1] = -1
+
+            if self.random.randint_inclusive(1, 100) != 1:
+                supware.append(w_id)
+            else:
+                supware.append(self.other_ware(w_id))
+                all_local = 0
+
+            qty.append(self.random.randint_inclusive(1, 10))
+
+        sql = 'SELECT new_order(%s, %s, %s, %s, %s, %s, %s, %s, %s)'
+        args = (w_id, c_id, d_id, order_line_count, all_local, itemid, supware, qty, timestamp)
+        return self.execute_sql(sql, args, dry_run)
+
+    def payment(self, timestamp, dry_run):
+        w_id = self.random.randint_inclusive(1, self.scale_factor)
+        d_id = self.random.randint_inclusive(1, DIST_PER_WARE)
+        c_id = self.random.nurand(1023, 1, CUST_PER_DIST)
+        h_amount = self.random.randint_inclusive(1, 5000)
+        c_last = self.tpcc_text.lastname(self.random.nurand(255, 0, 999))
+
+        byname = self.random.randint_inclusive(1, 100) <= 60
+        if self.random.randint_inclusive(1, 100) <= 85:
+            c_w_id = w_id
+            c_d_id = d_id
+        else:
+            c_w_id = self.other_ware(w_id)
+            c_d_id = self.random.randint_inclusive(1, DIST_PER_WARE)
+
+        sql = 'SELECT payment(%s, %s, %s, %s, %s, %s, %s, %s, %s)'
+        args = (w_id, d_id, c_d_id, c_id, c_w_id, h_amount, byname, c_last, timestamp)
+        return self.execute_sql(sql, args, dry_run)
+
+    def order_status(self, dry_run):
+        w_id = self.random.randint_inclusive(1, self.scale_factor)
+        d_id = self.random.randint_inclusive(1, DIST_PER_WARE)
+        c_id = self.random.nurand(1023, 1, CUST_PER_DIST)
+        c_last = self.tpcc_text.lastname(self.random.nurand(255, 0, 999))
+        byname = self.random.randint_inclusive(1, 100) <= 60
+
+        sql = 'SELECT * FROM order_status(%s, %s, %s, %s, %s)'
+        args = (w_id, d_id, c_id, c_last, byname)
+        return self.execute_sql(sql, args, dry_run)
+
+    def delivery(self, timestamp, dry_run):
+        w_id = self.random.randint_inclusive(1, self.scale_factor)
+        o_carrier_id = self.random.randint_inclusive(1, 10)
+
+        self.latest_delivery_date = timestamp
+
+        sql = 'SELECT * FROM delivery(%s, %s, %s, %s)'
+        args = (w_id, o_carrier_id, DIST_PER_WARE, timestamp)
+        return self.execute_sql(sql, args, dry_run)
+
+    def stock_level(self, dry_run):
+        w_id = self.random.randint_inclusive(1, self.scale_factor)
+        d_id = self.random.randint_inclusive(1, DIST_PER_WARE)
+        level = self.random.randint_inclusive(10, 20)
+
+        sql = 'SELECT * FROM stock_level(%s, %s, %s)'
```

Comment:
Not for this PR, but for the future. Sometimes we use this format to pass the arguments to the SQL and sometimes we use `f` strings. I think it'd be better to choose one of them as a standard and use it everywhere

Reply:
This is for `psycopg2`'s `execute` method, which only supports percent escapes.